### PR TITLE
add Promise caveat to AMD and require.ensure

### PR DESCRIPTION
**Pull request by jessehattabaugh**

### CODE REVIEW

File: content/api/module-methods.md [hunk 1]
```diff
@@ -169,6 +169,8 @@ require.ensure(dependencies: String[], callback: function(require), errorCallbac
 
 Split out the given `dependencies` to a separate bundle that that will be loaded asynchronously. When using CommonJS module syntax, this is the only way to dynamically load dependencies. Meaning, this code can be run within execution, only loading the `dependencies` if certain conditions are met.
 
+W> This feature relies on [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) internally. If you use `require.ensure` with older browsers, remember to shim `Promise` using a polyfill such as [es6-promise](https://github.com/stefanpenner/es6-promise) or [promise-polyfill](https://github.com/taylorhakes/promise-polyfill). See [Shimming](/guides/shimming) for more information.
```
Comment (simon04) [inline content/api/module-methods.md]: How does /guides/shimming give any better insight? I'd drop this sentence (here, below and for "`import()`").

File: content/api/module-methods.md [hunk 2]
```diff
@@ -246,6 +248,8 @@ require(dependencies: String[], [callback: function(...)])
 
 Similar to `require.ensure`, this will split the given `dependencies` into a separate bundle that will be loaded asynchronously. The `callback` will be called with the exports of each dependency in the `dependencies` array.
 
+W> This feature relies on [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) internally. If you use `import()` with older browsers, remember to shim `Promise` using a polyfill such as [es6-promise](https://github.com/stefanpenner/es6-promise) or [promise-polyfill](https://github.com/taylorhakes/promise-polyfill). See [Shimming](/guides/shimming) for more information.
```
Comment (simon04) [inline content/api/module-methods.md]: Ad "`import()`": copy/paste error?

**Comment by jessehattabaugh** [comment]
Sorry, I don't know what to do about the CLA. I used Github to make this edit, but Github wouldn't let me add another change to the same branch, so I cloned the repo on my work computer. jesseSB isn't a github user though. It's just the name that IT configured Git with. So I don't think I can sign the CLA with it.

**Comment by skipjack** [comment]
> Sorry, I don't know what to do about the CLA. I used Github to make this edit, but Github wouldn't let me add another change to the same branch, so I cloned the repo on my work computer. jesseSB isn't a github user though. It's just the name that IT configured Git with. So I don't think I can sign the CLA with it.

@jessehattabaugh yeah we've run into this a few times before as well. One approach would be:

1. `git reset 9fae5174c2b2f90e0a07f1a2b3017ab838483dbe` to the previous commit locally
2. re-stage and commit those things with the other account
3. `git push -f origin patch-2` to force update the branch

**Comment by simon04** [comment]
@jessehattabaugh I rebased your commits. The CLA should be fine now.

**Comment by simon04** [comment]
Some examples for later reference: https://github.com/webpack/webpack/tree/master/examples/code-splitted-require.context-amd, https://github.com/webpack/webpack/tree/master/examples/code-splitting-native-import-context

**Pull request merged**